src/webhooks/webhooks.ts

```diff
@@ -72,37 +71,64 @@ export class Webhooks {
     return [timestamp, signatureHash];
   }
 
-  computeSignature(timestamp: any, payload: any, secret: string): string {
+  async computeSignature(
+    timestamp: any,
+    payload: any,
+    secret: string,
+  ): Promise<string> {
     payload = JSON.stringify(payload);
     const signedPayload = `${timestamp}.${payload}`;
-    const expectedSignature = crypto
-      .createHmac('sha256', secret)
-      .update(signedPayload)
-      .digest()
-      .toString('hex');
 
-    return expectedSignature;
+    const key = await crypto.subtle.importKey(
+      'raw',
+      new TextEncoder().encode(secret),
+      { name: 'HMAC', hash: 'SHA-256' },
+      false,
+      ['sign'],
+    );
+
+    const signatureBuffer = await crypto.subtle.sign(
+      'HMAC',
+      key,
+      new TextEncoder().encode(signedPayload),
+    );
+
+    // crypto.subtle returns the signature in base64 format. This must be
+    // encoded in hex to match the CryptoProvider contract. We map each byte in
+    // the buffer to its corresponding hex octet and then combine into a string.
+    const signatureBytes = new Uint8Array(signatureBuffer);
+    const signatureHexCodes = new Array(signatureBytes.length);
+
+    for (let i = 0; i < signatureBytes.length; i++) {
+      signatureHexCodes[i] = byteHexMapping[signatureBytes[i]];
+    }
+
+    return signatureHexCodes.join('');
```

Comment:
Most of this was copied over from stripe: https://github.com/stripe/stripe-node/blob/master/src/crypto/SubtleCryptoProvider.ts